src/overlays/actors/ovl_En_Rd/z_en_rd.c

```diff
@@ -85,126 +133,1137 @@ static DamageTable D_808D710C = {
+    } else if (this->unk_3E4 < 1000) {
+        if ((this->actor.params != ENRD_GET_2) && (this->unk_3EC == 0)) {
+            func_808D5C54(this);
+        } else {
+            func_808D6008(this);
+        }
+    }
+}
+
+void func_808D4DC4(EnRd* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600ABE0, 0.0f, 0.0f, SkelAnime_GetFrameCount(&D_0600ABE0.common), 0,
+                         -6.0f);
+    this->unk_3EF = 13;
+    this->unk_3D6 = 6;
+    this->actor.shape.rot.x = -0x4000;
+    this->actor.gravity = 0.0f;
+    this->actor.shape.yOffset = 0.0f;
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = func_808D4E60;
+}
+
+void func_808D4E60(EnRd* this, GlobalContext* globalCtx) {
+    if (this->actor.shape.rot.x != -0x4000) {
+        Math_SmoothStepToS(&this->actor.shape.rot.x, 0, 1, 0x7D0, 0);
+        if (Math_SmoothStepToF(&this->actor.world.pos.y, this->actor.home.pos.y, 0.3f, 2.0f, 0.3f) == 0.0f) {
+            this->actor.gravity = -3.5f;
+            func_808D4308(this);
+        }
+    } else {
+        if (this->actor.world.pos.y == this->actor.home.pos.y) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_CRY);
+        }
+
+        if (Math_SmoothStepToF(&this->actor.world.pos.y, this->actor.home.pos.y + 50.0f, 0.3f, 2.0f, 0.3f) == 0.0f) {
+            if (this->unk_3D6 != 0) {
+                this->unk_3D6--;
+                Math_SmoothStepToF(&this->actor.speedXZ, 6.0f, 0.3f, 1.0f, 0.3f);
+            } else if (Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 0.3f, 1.0f, 0.3f) == 0.0f) {
+                Math_SmoothStepToS(&this->actor.shape.rot.x, 0, 1, 0x7D0, 0);
+            }
+        }
+    }
+}
+
+void func_808D4FE0(EnRd* this, GlobalContext* globalCtx) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_060113EC.common);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060113EC, 1.0f, 4.0f, frameCount, 1, -4.0f);
+    this->actor.speedXZ = 0.4f;
+    this->unk_3EF = 4;
+    this->actionFunc = func_808D506C;
+}
+
+void func_808D506C(EnRd* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
+    s16 sp36 = ((this->actor.yawTowardsPlayer - this->actor.shape.rot.y) - this->unk_3D8) - this->unk_3DA;
+
+    this->skelAnime.animPlaybackSpeed = this->actor.speedXZ;
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 250, 0);
+    Math_SmoothStepToS(&this->unk_3D8, 0, 1, 100, 0);
+    Math_SmoothStepToS(&this->unk_3DA, 0, 1, 100, 0);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if (Actor_DistanceToPoint(&player->actor, &this->actor.home.pos) >= 150.0f) {
+        func_808D53C0(this, globalCtx);
+    }
+
+    if ((ABS_ALT(sp36) < 0x1554) && (Actor_DistanceBetweenActors(&this->actor, &player->actor) <= 150.0f)) {
+        if (!(player->stateFlags1 & 0x2C6080) && !(player->stateFlags2 & 0x4080)) {
```

Comment:
```suggestion
        if (!(player->stateFlags1 & (0x200000 | 0x80000 | 0x40000 | 0x4000 | 0x2000 | 0x80)) && !(player->stateFlags2 & 0x4080)) {
```
I'm really surprised how many bits they packed into one number. While separating it out like this may seem a little obnoxious, I think I'd still prefer it, especially once each flag for stateFlags1 is known